### PR TITLE
Tweaked prompt and added unit-tests for history tutor. Some UI fixes to history tutor agent.

### PR DESCRIPTION
**Pull request by adamconnors**

### CODE REVIEW

File: py-server/history_tutor/history_tutor_test.py [hunk 1]
```diff
@@ -1,51 +1,255 @@
...
+
+def chat(tutor, message_history, latest_message, world_state):
+    message_history = build_message_history(message_history)
+    initial_agent_state = AgentState(latest_message, message_history, world_state)
+    updated_agent_state = tutor.update_state(initial_agent_state)
+    response = tutor.chat(latest_message)
+    return response, updated_agent_state
+
+
+# TODO: What's the proper way to do this, bool didn't seem to work?
```
Comment (alalamav) [inline py-server/history_tutor/history_tutor_test.py:239]: the bool constructor treats strings as truthy, ie. `bool("") == False`, `bool("false") == True`

File: py-server/history_tutor/history_tutor.py [hunk 2]
```diff
@@ -67,10 +66,10 @@
         {{
             "question": "What were the different theories about the cause of the Black Death?",
             "answers": [
-            {{"answer": "Religion: God sent the plague as a punishment for people's sins.", "hasAnswered": false}},
-            {{"answer": "Miasma: bad air or smells caused by decaying rubbish.", "hasAnswered": false}},
-            {{"answer": "Four Humours: most physicians believed that disease was caused by an imbalance in the Four Humours.", "hasAnswered": false}},
-            {{"answer": "Outsiders: strangers or witches had caused the disease.", "hasAnswered": false}}
+            {{"answer": "Religion: God sent the plague as a punishment for people's sins.", "hasAnswered": "false"}},
+            {{"answer": "Miasma: bad air or smells caused by decaying rubbish.", "hasAnswered": "false"}},
```
Comment (alalamav) [inline py-server/history_tutor/history_tutor.py:70]: I believe that providng detailed examples may risk overfitting the model's generated questions. I have been experimenting with only providing the (typescript) interfaces to the model, without examples, and it seems to work fine.

Are you seeing value in providing few-shot?

**Review by alalamav — APPROVED** [review]
left some questions about prompts, but LGTM

File: py-server/history_tutor/history_tutor.py [hunk 3]
```diff
@@ -67,10 +66,10 @@
         {{
             "question": "What were the different theories about the cause of the Black Death?",
             "answers": [
-            {{"answer": "Religion: God sent the plague as a punishment for people's sins.", "hasAnswered": false}},
-            {{"answer": "Miasma: bad air or smells caused by decaying rubbish.", "hasAnswered": false}},
-            {{"answer": "Four Humours: most physicians believed that disease was caused by an imbalance in the Four Humours.", "hasAnswered": false}},
-            {{"answer": "Outsiders: strangers or witches had caused the disease.", "hasAnswered": false}}
+            {{"answer": "Religion: God sent the plague as a punishment for people's sins.", "hasAnswered": "false"}},
```
[the remark this answers is not in the document]
Comment (adamconnors) [reply py-server/history_tutor/history_tutor.py:69]: I was getting parse errors without them -- but of course, the serialized JSON that was being produced doesn't always match the examples. 

I need to take a step back & revisit this in general.

File: py-server/history_tutor/history_tutor_test.py (hunk 1 above)
Comment (adamconnors) [reply py-server/history_tutor/history_tutor_test.py:239]: make sense (kind of) but when what's the right way to go from a string (true/false) to a boolean? or is it just to do a string comparison as I have here?

**Review by adamconnors — COMMENTED** [review]
Thanks Alberto ... as you've pointed out, I've learned some lessons trying to do it this way. I think I need to take a step back & have another stab. I'll merge this for posterity, but come back with another iteration.

**Pull request merged**